src/views/Workflow.vue

```diff
@@ -72,24 +98,49 @@ export default {
      * @type {CylcTree}
      */
     tree: new CylcTree(),
-    isLoading: true
+    isLoading: true,
+    // the widgets added to the view
+    /**
+     * @type {
+     *   Object.<string, string>
+     * }
+     */
+    widgets: {}
```

Comment:
A single map, so that we don't need one function for each delete event. We could have one single function for the add event too, but our components have different "signatures" (the props passed down, as well as events, slots, etc).

In the future we should be able to use this `widgets` Object, without the need to have `treeWidgets`, `graphWidgets`, `mutationsWidgets`, `etcWidgets`. But that's for a future PR :+1: